HuntTheWumpus/HuntTheWumpus.cpp

```diff
@@ -201,66 +200,90 @@ int main()
 
     while (change.IsPlaying())
     {
+
         // Parse input.
         std::string input;
 
         std::cout << "Command? " << std::flush;
 
-        std::getline(std::cin, input);
-
-        // Split into strings.
-        const auto stringTokens = SplitString(input, " \t\n");
-
-        const auto command = ToLower(stringTokens[0]);
-
-        if (command[0] == 'm')
+        try
         {
-            if (stringTokens.size() < 2)
+            std::getline(std::cin, input);
+            // Split into strings.
+            const auto stringTokens = SplitString(input, " \t\n");
+
+            if (stringTokens.size() == 0)
             {
-                std::cout << "A Move command must be followed by the destination cave id.\n";
-                continue;
+                throw std::runtime_error("");
             }
 
-            // Second token is a destination.
-            const auto destCave = std::stoi(stringTokens[1]);
-
-            dungeon.MakeMove(HuntTheWumpus::DungeonMove::Move, { destCave });
-        }
+            const auto command = ToLower(stringTokens[0]);
 
-        if (command[0] == 's')
-        {
-            if (stringTokens.size() < 2)
+            if (command[0] == 'm' && command.length() != 0)
```

Comment:
Your logic is inverted. If we access an empty command (and it's not obvious how we can take a string that's not whitespace and end up with an empty token) then you need to check that it's not empty before dereferencing the 0th element.

It's also better to use .empty() than .length() != 0 or .size() != 0.

```
if(command.empty())
{
    std::cout << "Could not understand the command " << input << std::endl;
    continue;
}

if(command[0] == 'm') 
```

Think also about how you could possible manifest command.length() == 0 or command.empty(). What possible input could you generate that would cause  this?